src/domain/accounts/domain/src/entities/account.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, TimeZone, Utc};
+use lazy_static::lazy_static;
+use opendatafabric::{AccountID, AccountName};
+use serde::{Deserialize, Serialize};
+
+///////////////////////////////////////////////////////////////////////////////
+
+// TODO: have some length restrictions (0 < .. < limit)
+pub type AccountDisplayName = String;
+
+pub const DEFAULT_ACCOUNT_NAME_STR: &str = "kamu";
+
+lazy_static! {
+    pub static ref DEFAULT_ACCOUNT_NAME: AccountName =
+        AccountName::new_unchecked(DEFAULT_ACCOUNT_NAME_STR);
+    pub static ref DEFAULT_ACCOUNT_ID: AccountID =
+        AccountID::new_seeded_ed25519(DEFAULT_ACCOUNT_NAME_STR.as_bytes());
+    static ref DUMMY_REGISTRATION_TIME: DateTime<Utc> =
+        Utc.with_ymd_and_hms(2024, 4, 1, 12, 0, 0).unwrap();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Account {
+    pub id: AccountID,
+    pub account_name: AccountName,
+    pub email: Option<String>,
+    pub display_name: AccountDisplayName,
+    pub account_type: AccountType,
+    pub provider: AccountProvider,
+    pub avatar_url: Option<String>,
+    pub registered_at: DateTime<Utc>,
+    pub is_admin: bool,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum AccountType {
+    User,
+    Organization,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum AccountProvider {
+    Cli,
+    Github(AccountProviderGithub),
```

Comment:
(Related to my concern with having a dedicated `github_accounts` table)

Approaching providers this way means that adding a new provider is a domain-level change. A person adding a provider will have to add a migration and add an enum variant here etc. If they are adding a non-mainstream provider for their own org - their changes cannot be upstreamed and they are stuck maintaining a complete fork.

Such lack of extensibility is a sucky experience - I just went through it with RisingWave engine.

Ideally if I need to add a custom auth provider - I should be able to link `kamu` libraries and just throw my custom provider implementation into `Catalog`. No migrations, no patching code.

We already had auth providers identified via [string IDs](https://github.com/kamu-data/kamu-cli/blob/755ec857fb77f420069b4b3e84dd65fea396ae48/src/adapter/oauth/src/oauth_github.rs#L225), and I liked this model.

Reply:
Another possibility is to externalize additional provider-specific data, without coupling with the account. It might live in separate table of it's own with it's own repositoriy and associate some provider-specific data with `AccountID`.

Reply:
I think I've demonstrated my approach with separate tables looks quite neat.